### ISSUE TITLE
Generate Report of Student Submission

### ISSUE THREAD

**Issue by soachishti** [BilalZaib/AutoGrader#6]

**Comment by recluze** [comment]
This would be higher priority than #9 ...

**Comment by recluze** [comment]
Please push to branch whenever you have something on reports so that I can take a look. I promise I won't mess with your branch =]

**Comment by soachishti** [comment]
Sure, I will keep the remote branch updated. =]

**Comment by soachishti** [comment]
Pull request added. #23

**Comment by recluze** [comment]
Great work with the report. I really like the "Edit" button to fix any mistakes by the autograder. 

Some changes: 

1. Should add the "username" column along with roll no. 
2. Add "Submission date". This would be for the latest ... just as the pass/fail fields. 
3. Make table sortable on all columns (client-side). 
4. For "percent", I forget: are we actually displaying "percent" or total points (given that we have assignment score in the meta data for the assignment. I think we should be displaying actual "score" instead of percentage. 

BTW, did you check with multiple students/assignments to ensure we're not getting any mixups?

**Comment by recluze** [comment]
(very low priority: syntax highlighting for the code shown in the popup in report.)

**Comment by soachishti** [comment]
#24

**Comment by recluze** [comment]
Assignment report seems to have a bug: 

```
def get_student_latest_submissions(self):
        # TODO: Try to acheive this by sub queries using Django ORM
        # Get students of this course
        students = Student.objects.filter(courses=self.course)
        submissions = []
        for student in students:
            submission = Submission.objects.filter(student=student).order_by("-publish_date").first()
            if submission:
                submissions.append(submission)
``` 

It's not checking for a particular assignment. We're getting back whatever happens to be the latest submission. So, if you submit an older assignment later, that shows up.

**Comment by recluze** [comment]
The way we need to do this is to find the course associated with the assignment, then find all students that are enrolled and then get their score for this particular assignment. In case there is no record of submission by a student, we show a `-` in their row. This is so that we know exactly which students have submitted and not omit those that haven't made any submission.

**Comment by soachishti** [comment]
Such a silly mistake :) #27

**Comment by recluze** [comment]
Nice work with the report. It seems very usable now. 

Just get rid of the duplicate `assignment` object from here: https://github.com/BilalZaib/AutoGrader/blob/8ae7e29e9fbc94d2d8c10be1de887a9b8a2a72f2/AutoGrade/views.py#L334

And see if you can disable email verification on the other bug. We can then deploy this tomorrow inshaallah and see if the students can bring up some bugs. 

Thanks again for all the good work (especially over the holidays).

**Comment by soachishti** [comment]
Fixed the duplicate in #31 and Email verification is fixed in #30 using Outlook SMTP.

**Comment by recluze** [comment]
Let's close this.
